salt/grains/core.py

```diff
@@ -833,21 +833,22 @@ def _virtual(osdata):
                 break
         elif command == "virt-what":
             try:
-                output = output.splitlines()[-1]
+                output = output.splitlines()[0]
             except IndexError:
                 pass
-            if output in ("kvm", "qemu", "uml", "xen", "lxc"):
-                grains["virtual"] = output
-                break
-            elif "vmware" in output:
-                grains["virtual"] = "VMware"
-                break
-            elif "parallels" in output:
-                grains["virtual"] = "Parallels"
-                break
-            elif "hyperv" in output:
-                grains["virtual"] = "HyperV"
-                break
+            for output in output.splitlines():
```

Comment:
I think it might be better to use a different variable name that does not overwrite the value of `output`.
```python
>>> output = "foo\nbar"                                                                                                                                
>>> for output in output.splitlines():                                                                                                                 
...     pass                                                                                                                                           
...                                                                                                                                                    
>>> print(output)                                                                                                                                      
bar 
```
for example:
```diff
- for output in output.splitlines():
+ for line in output.splitlines():
```